dashboard/app/views/devise/registrations/edit.html.haml

```diff
@@ -69,12 +73,13 @@
       = f.select :user_type, user_type_options
 
   %div= f.submit t('crud.update'), class: 'btn btn-default'
-%h3= t('user.delete')
-%p
-  = t('user.delete_text')
-  - if current_user.teacher?
-    = t('user.delete_text_teacher_note')
-  = button_to t('user.delete'), registration_path(resource_name), data: { confirm: t('user.delete_confirm') }, method: :delete, class: 'btn btn-danger'
+- if current_user.can_delete_account?
```

Comment:
Possibly, but I'm a little bit nervous about changing it just to change it - you can't use `f.object` here because it's not in scope for that code, and while it does seem like wherever `f.object` above should be the same as `current_user`, it's coming in as a param to the form block, so technically that would be changing the semantics. While I don't *think* it should make a difference, my (very slight) preference would be to not touch the existing code if it's not broken. Happy to change the `f.object` uses if you feel strongly about it, though.